components/org.wso2.carbon.identity.governance/src/test/java/org/wso2/carbon/identity/governance/listener/IdentityStoreEventListenerTest.java

```diff
@@ -0,0 +1,329 @@
+                return null;
+            }
+        }).when(userIdentityDataStore).store(userIdentityClaim, userStoreManager);
+
+        boolean doPreSetClaimValues = identityStoreEventListener.doPreSetUserClaimValues(userName, claims, prof, userStoreManager);
+        assertTrue("Do pre set claim values is invalid.", doPreSetClaimValues);
+    }
+
+    @Test(dataProvider = "dataHandler2")
+    public void testDoPostGetUserClaimValues(String userName, Object pwd, String[] claimList, Map<String, String> claims,
+                                             String prof) throws UserStoreException,
+            NoSuchFieldException, IllegalAccessException, IdentityException {
+        realmConfiguration = mock(RealmConfiguration.class);
+        userIdentityDataStore = mock(UserIdentityDataStore.class);
+
+        Field fieldIdentityStore = IdentityStoreEventListener.class
+                .getDeclaredField("identityDataStore");
+        fieldIdentityStore.setAccessible(true);
+        fieldIdentityStore.set(identityStoreEventListener, userIdentityDataStore);
+
+        boolean postGetUserClaims = identityStoreEventListener.doPostGetUserClaimValues(userName, claimList, prof,claims, userStoreManager);
+        assertTrue(postGetUserClaims);
+    }
+
+    @Test(dataProvider = "dataHandler3")
+    public void testDoPreGetUserClaimValue(String userName, String claim, String profileName ) throws UserStoreException {
+        boolean preGetUserClaims = identityStoreEventListener.doPreGetUserClaimValue(userName, claim, profileName, userStoreManager);
+        assertTrue(preGetUserClaims);
+    }
+    @Test(expectedExceptions = UserStoreException.class)
+    public void testDoPreGetUserClaimValueException() throws UserStoreException {
+
+        boolean preGetUserClaims = identityStoreEventListener.doPreGetUserClaimValue("admin",
+                "http://wso2.org/claims/identity/email", "myprofile", userStoreManager);
+        assertTrue(preGetUserClaims);
+    }
+
+    @Test(dataProvider = "dataHandler4")
+    public void testDoPreSetUserClaimValue(String userName, String claimUri, String claimValue, String profileName)
+            throws UserStoreException {
+        boolean preGetUserClaims = identityStoreEventListener.doPreSetUserClaimValue(userName, claimUri,
+                claimValue, profileName, userStoreManager);
+        assertTrue(preGetUserClaims);
+    }
+
+    @Test(expectedExceptions = UserStoreException.class)
+    public void testDoPreSetUserClaimValueException() throws UserStoreException {
+        boolean preGetUserClaims = identityStoreEventListener.doPreSetUserClaimValue("admin",
+                "http://wso2.org/claims/identity/email", "admin@wso2.com", "foo", userStoreManager);
+        assertTrue(preGetUserClaims);
+    }
+
+    @Test(dataProvider = "dataHandler5")
+    public void testDoPreGetUserList(String claimUri, String claimValue, final List<String> userList,
```

Comment:
It is fine to declare throws Exception rather than defining specific Exception classes for tests?